### PR TITLE
simplify complex atanh and remove singularity perturbation

### LINKED ISSUE #55266: Complex atanh is inaccurate near singularities
`real(atanh(1+y*im))` should be asymptotic to `log(2/abs(y))/2` for real `y` as `y` approches `0`.

For example, this suggests that `real(atanh(1+1e-200im))` should evaluate to `log(2/1e-200)/2=230.60508288968455`, but it currently evaluates to `177.09910463306602`, which has no correct digits.

This appears to be caused by a small positive perturbation, `ρ`, that is added to the imaginary part of the input in the atanh computation https://github.com/JuliaLang/julia/blob/1dee0006c453874a1f8965d55a27befc2620f253/base/complex.jl#L1040-L1042 ... https://github.com/JuliaLang/julia/blob/1dee0006c453874a1f8965d55a27befc2620f253/base/complex.jl#L1067-L1068

so that `real(atanh(1+1e-200im))` in fact computes `log(2/ρ)=177.09910463306602`.

This perturbation is suggested by [Kahan 86](https://people.freebsd.org/~das/kahan86branch.pdf), but as far as I can tell, it is not necessary in either of the places it is used, and in fact harms rather than helps accuracy when the real part of the input is `1`. It may be there to avoid division by 0 in `atanh(1.0+0.0im)`, but the Julia implementation already handles this case with an explicit branch.

For comparison, [cpython](https://github.com/python/cpython/blame/bc94cf7e254e43318223553a7959115573c679a5/Modules/cmathmodule.c#L366-L412) computes the correct answer:
```
> python3
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import cmath
>>> cmath.atanh(1+1e-200j)
(230.60508288968455+0.7853981633974483j)
```

Version Info:
```
julia> versioninfo()
Julia Version 1.10.4
Commit 48d4fd48430 (2024-06-04 10:41 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 12 × Apple M2 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)
```

### PR DESCRIPTION
**Pull request by oscardssmith**
fixes https://github.com/JuliaLang/julia/issues/55266, and use `inv(z)` rather than `1/z` and use `muladd` in a couple places.

### CODE REVIEW

File: base/complex.jl [hunk 1]
```diff
@@ -1064,16 +1062,16 @@ function atanh(z::Complex{T}) where T
             ξ = oftype(x, Inf)
             η = y
         else
-            ym = ay+ρ
-            ξ = log(sqrt(sqrt(4+y*y))/sqrt(ym))
+            ym = ay
+            ξ = log(sqrt(sqrt(muladd(y, y, 4)))/sqrt(ym))
             η = copysign(oftype(y,pi)/2 + atan(ym/2), y)/2
```
[the remark this answers is not in the document]
Comment (oscardssmith) [reply base/complex.jl]: will add tests. I mostly wanted to make sure that it passed existing tests first.

[the remark this answers is not in the document]
Comment (giordano) [reply base/complex.jl]: Looks like existing tests are passing

**Comment by oscardssmith** [comment]
tests added

File: test/complex.jl [hunk 2]
```diff
@@ -1215,3 +1215,9 @@ end
     @test !iseven(7+0im) && isodd(7+0im)
     @test !iseven(6+1im) && !isodd(7+1im)
 end
+
+@testset "issue #55266" begin
+    for T in (Float32, Float64)
```
Comment (giordano) [inline test/complex.jl:1220]:
```suggestion
    for T in (Float16, Float32, Float64)
```
?

Comment (oscardssmith) [reply test/complex.jl]: might as well

**Pull request merged**